src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/OntologyValidator.java

```diff
@@ -83,6 +82,29 @@ protected void checkTermUniqueness(String termName, Integer id, String name, Int
 		}
 	}
 
+	protected void checkTermUniqueness(final String fieldName, final String propertyName, final Integer id, final String name,
+		final Integer cvId, final Errors errors) {
+
+		try {
+			Term term = this.termDataManager.getTermByNameAndCvId(name, cvId);
+			if (term == null) {
+				return;
+			}
+
+			if (Objects.equals(id, null) && Objects.equals(term, null)) {
+				return;
+			}
+
+			if (id != null && Objects.equals(id, term.getId())) {
+				return;
+			}
+
+			this.addCustomError(errors, fieldName, BaseValidator.NAME_OR_ALIAS_ALREADY_EXIST, new Object[] {fieldName, propertyName});
+		} catch (MiddlewareException e) {
+			OntologyValidator.LOGGER.error("Error checking uniqueness of term name", e);
+		}
+	}
```

Comment:
it seems there is a `this.addDefaultError(errors);` in that class that you can use inside the catch.
Though that class seem to be quite old, usually what we do now is to rethrow. You can also rethrow either ApiRequestValidationException or ApiValidationException
Note aside, I'm migrating the class ApiRuntimeException in IBP-4732, to be able to handle error codes instead of plain text messages. I'll push in a separate commit when it's ready

Reply:
done.